src/preset.ts

```diff
@@ -1180,7 +1055,64 @@ async function tryApplyVsDevEnv(preset: ConfigurePreset, workspaceFolder: string
     preset.__parentEnvironment = EnvironmentUtils.mergePreserveNull([process.env, developerEnvironment]);
 }
 
-async function expandConfigurePresetHelper(folder: string, preset: ConfigurePreset, allowUserPreset: boolean = false, errorHandler?: ExpansionErrorHandler) {
+export async function expandConfigurePreset(folder: string, name: string, workspaceFolder: string, sourceDir: string, allowUserPreset: boolean = false, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler): Promise<ConfigurePreset | null> {
+
+    // TODO: We likely need to refactor to include these refs, for configure, build, test, etc Presets.
+    const refs = referencedConfigurePresets.get(folder);
+    if (!refs) {
+        referencedConfigurePresets.set(folder, new Set());
+    } else {
+        refs.clear();
+    }
+
+    // if enableTryApplyDevEnv is true, we have to check if the preset is valid in expandedPresetsFiles
```

Comment:
Remind me what case this is protecting against?

Reply:
making sure that the preset we are setting is valid in the expanded presets map, since expansion will be using the original presets map which is pre validation